lib/readline.js

```diff
@@ -397,8 +401,11 @@ Interface.prototype.question[promisify.custom] = function(query, options) {
   options = typeof options === 'object' && options !== null ? options : {};
 
   return new Promise((resolve, reject) => {
-    this.question(query, options, resolve);
+    if (options.signal?.aborted) {
```

Comment:
How far back is this expected to be backported?

Also I would prefer if this check was one line above and just `return Promise.reject(` since throwing inside the promise constructor is not very fun ^^

Reply:
done